server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -988,13 +1048,50 @@ private void asyncCleanupUnlinkedShardLevelBlobs(
         }
     }
 
-    private void executeStaleShardDelete(BlockingQueue<List<String>> staleFilesToDeleteInBatch, GroupedActionListener<Void> listener)
-        throws InterruptedException {
+    private void executeStaleShardDelete(
```

Comment:
Lets add the comment to explain the difference when `remoteStoreLockManagerFactory` is `null` and `non-null`